File: CS/CustomizeReportDataExample.Module.Win/ReportWizardModifyController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.ReportsV2.Win;
using CustomizeReportDataExample.Module.BusinessObjects;


namespace CustomizeReportDataExample.Module.Win {
    public class ReportWizardModifyController : ViewController {
        WinReportServiceController reportServiceController;
        public ReportWizardModifyController() { }
        protected override void OnActivated() {
            base.OnActivated();
            reportServiceController = Frame.GetController<WinReportServiceController>();
            reportServiceController.NewXafReportWizardShowing +=
                reportServiceController_NewXafReportWizardShowing;
        }
        protected override void OnDeactivated() {
            reportServiceController.NewXafReportWizardShowing -=
                reportServiceController_NewXafReportWizardShowing;
            reportServiceController = null;
            base.OnDeactivated();
        }
        void reportServiceController_NewXafReportWizardShowing(object sender,
            NewXafReportWizardShowingEventArgs e) {
            if (!e.ReportDataType.Equals(typeof(MyReportDataV2))) return;
            CustomReportWizardParameters newReportParamsObject = new
                CustomReportWizardParameters(e.WizardParameters.Report, e.WizardParameters.ReportDataType);
            newReportParamsObject.Category = "Default";
            e.WizardParameters = newReportParamsObject;
        }
    }
}

File: VB/CustomizeReportDataExample.Module.Win/ReportWizardModifyController.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.ReportsV2.Win
Imports CustomizeReportDataExample.Module.BusinessObjects


Namespace CustomizeReportDataExample.Module.Win
    Public Class ReportWizardModifyController
        Inherits ViewController

        Private reportServiceController As WinReportServiceController
        Public Sub New()
        End Sub
        Protected Overrides Sub OnActivated()
            MyBase.OnActivated()
            reportServiceController = Frame.GetController(Of WinReportServiceController)()
            AddHandler reportServiceController.NewXafReportWizardShowing, AddressOf reportServiceController_NewXafReportWizardShowing
        End Sub
        Protected Overrides Sub OnDeactivated()
            RemoveHandler reportServiceController.NewXafReportWizardShowing, AddressOf reportServiceController_NewXafReportWizardShowing
            reportServiceController = Nothing
            MyBase.OnDeactivated()
        End Sub
        Private Sub reportServiceController_NewXafReportWizardShowing(ByVal sender As Object, ByVal e As NewXafReportWizardShowingEventArgs)
            If Not e.ReportDataType.Equals(GetType(MyReportDataV2)) Then
                Return
            End If
            Dim newReportParamsObject As New CustomReportWizardParameters(e.WizardParameters.Report, e.WizardParameters.ReportDataType)
            newReportParamsObject.Category = "Default"
            e.WizardParameters = newReportParamsObject
        End Sub
    End Class
End Namespace
